eng/common/scripts/modules/git-api-calls.psm1

```diff
@@ -0,0 +1,81 @@
+# Methods for interacting with the github or azure repo APIs
+# Retrieves the list of all tags that exist on the target repository
+# Changes 1: Pass full API URL instead of formulating it within the method
+# Changes 2: Extend method for git Repos
+function GetExistingTags($apiUrl, $repoType = "github", $headers = $null) {
+    try {
+      if ($repoType -eq "github")
+      {
+        return (Invoke-RestMethod -Method "GET" -Uri $apiUrl  ) | % { $_.ref.Replace("refs/tags/", "") }
+      }
+      else {
+        return (Invoke-RestMethod -Method "GET" -Uri $apiUrl -Headers $headers ) | % { $_.value.name.Replace("refs/tags/", "") }
```

Comment:
Shouldn't this use FireAPIRequest instead of directly calling Invoke-RestMethod?